test/ssltest.c

```diff
@@ -454,8 +454,8 @@ static int verify_alpn(SSL *client, SSL *server)
     OPENSSL_free(alpn_selected);
     alpn_selected = NULL;
 
-    if (client_proto_len != server_proto_len ||
-        memcmp(client_proto, server_proto, client_proto_len) != 0) {
+    if (client_proto != NULL && (client_proto_len != server_proto_len ||
```

Comment:
As far as the test cases go, `server_proto` is never NULL if `client_proto != NULL`.

Though for the test sake, the check should probably make sure that if `client_proto == NULL` then also `server_proto == NULL`, otherwise it should fail.